client/my-sites/site-settings/analytics/form-google-analytics-jetpack.jsx

```diff
@@ -90,6 +91,25 @@ const GoogleAnalyticsJetpackForm = ( {
 		handleSubmitForm();
 	};
 
+	const handleSettingsToggleChange = ( value ) => {
+		recordTracksEvent( 'calypso_google_analytics_setting_changed', { key: 'is_active', path } );
```

Comment:
I wasn't seeing the tracks events being fired, and I recalled from working with `wp-calypso` that there are two versions of `recordTracksEvents` used throughout the codebase. The imported version in this file is dispatched everywhere I've looked at so far ( `dispatch( recordTracksEvent..`).

Reply:
Can you please also check if they appear in console after setting `localStorage.debug = 'calypso:analytics'`.
Looks like Calypso doesn't track events on dev instances, but they should appear in the browser's console.

Reply:
Since this would now be live, I retested checking in WordPress.com itself. I see the same - I don't see the `calypso_google_analytics_setting_changed` tracking event (checking the network tab, Tracks Vigilante (extension), and the MC Tracks live view. What I do see on toggle change is both the `calypso_jetpack_module_toggle` event and the `calypso_jetpack_module_toggle_google-analytics_on` (or `calypso_jetpack_module_toggle_google-analytics_off` events being fired though ([source](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/site-settings/jetpack-module-toggle.jsx#L59)).

That said, since that toggle off and toggle on event exists already, would we need another? The existing event also includes the path.

Reply:
Hi @coder-karen 👋 
You are right, I wasn't testing it correctly, so there is a problem with testing. Good catch 👍 

I created a PR to address the issues and improve the code a bit, please take a look: https://github.com/Automattic/wp-calypso/pull/92564

> That said, since that toggle off and toggle on event exists already, would we need another?

The event `calypso_google_analytics_setting_changed` has been there for years for other GA settings. I only expand its usage to track the `is_active` setting changes.
The `module_toggle` event won't get triggered when we remove Jetpack module. `calypso_seo_settings_google_analytics_updated` will be triggered instead, but it doesn't include the settings keys that get changed.

So I think using `calypso_google_analytics_setting_changed` makes sense, as it'll be the only single event to track both types of Google Analytics activation.

Reply:
Got it, makes sense.